### PR TITLE
[Android] AR mode switch, fixes

### PR DESCRIPTION
**Pull request by kiryldz**
- Add simple button for AR modes switch
- Fix of quite common crash because of `lateinit var navigation: MapboxNavigation`
- Use normal click in Replayer AR map instead of long click

### CODE REVIEW

File: app/src/main/java/com/mapbox/vision/examples/activity/ar/ArNavigationActivity.kt [hunk 1]
```diff
@@ -67,6 +68,7 @@ class ArNavigationActivity : AppCompatActivity(), RouteListener, ProgressChangeL
     private lateinit var routeFetcher: RouteFetcher
 
     private var lastRouteProgress: RouteProgress? = null
+    private var activeArFeature: ArFeature = ArFeature.Lane
```
Comment (yunikkk) [inline app/src/main/java/com/mapbox/vision/examples/activity/ar/ArNavigationActivity.kt]: Does it mean only Lane is shown by default? Seems it's better to turn on everything to allow users to see new shiny Fence from the beginning

Comment (kiryldz) [reply app/src/main/java/com/mapbox/vision/examples/activity/ar/ArNavigationActivity.kt]: I sticked to our old discussion stating that default parameters for `VisionArView` are `fenceVisiblity=false && laneVisibility=true`.
However I indeed like more an idea to turn everything on as default.
We'll return to that as soon as UI will be ready for the switch.

File: common/src/main/java/com/mapbox/vision/common/models/ArFeature.kt [hunk 2]
```diff
@@ -0,0 +1,31 @@
+package com.mapbox.vision.common.models
+
+sealed class ArFeature {
+
+    open val caption: String = ""
```
Comment (yunikkk) [inline common/src/main/java/com/mapbox/vision/common/models/ArFeature.kt]: Can this 3 vals just be params in constructor, should be shorter then, eg.
```
sealed class ArFeature(
  val caption: String,
  ...
) {
  object Lane : ArFeature(
     caption = "blabla",
     ...
  )
}
```

Comment (kiryldz) [reply common/src/main/java/com/mapbox/vision/common/models/ArFeature.kt]: Yeah, reasonable.

File: app/src/main/java/com/mapbox/vision/examples/activity/ar/ArNavigationActivity.kt (hunk 1 above)
Comment (kiryldz) [reply app/src/main/java/com/mapbox/vision/examples/activity/ar/ArNavigationActivity.kt]: @yunikkk - leaving `Lane` mode as default.

Comment (kiryldz) [reply app/src/main/java/com/mapbox/vision/examples/activity/ar/ArNavigationActivity.kt]: @yunikkk - changed my mind - we will have enabled both `Lane` and `Fence` at start.

**Pull request merged**